From 4f6797017b45624bcbdbf0e158482ee1b93985a1 Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Sat, 7 Aug 2021 11:11:39 +0200
Subject: [PATCH 40/40] When aux cameras are enabled, ignore "system only"
 camera flag (it shouldnt be q security issue since secure cameras are listed
 otherwise)

---
 .../libcameraservice/common/CameraProviderManager.cpp  | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/services/camera/libcameraservice/common/CameraProviderManager.cpp b/services/camera/libcameraservice/common/CameraProviderManager.cpp
index b7d1a57567..75748e8b15 100644
--- a/services/camera/libcameraservice/common/CameraProviderManager.cpp
+++ b/services/camera/libcameraservice/common/CameraProviderManager.cpp
@@ -618,10 +618,12 @@ SystemCameraKind CameraProviderManager::ProviderInfo::DeviceInfo3::getSystemCame
 
     // Go through the capabilities and check if it has
     // ANDROID_REQUEST_AVAILABLE_CAPABILITIES_SYSTEM_CAMERA
-    for (size_t i = 0; i < entryCap.count; ++i) {
-        uint8_t capability = entryCap.data.u8[i];
-        if (capability == ANDROID_REQUEST_AVAILABLE_CAPABILITIES_SYSTEM_CAMERA) {
-            return SystemCameraKind::SYSTEM_ONLY_CAMERA;
+    if(!property_get_bool("persist.sys.phh.include_all_cameras", false)) {
+        for (size_t i = 0; i < entryCap.count; ++i) {
+            uint8_t capability = entryCap.data.u8[i];
+            if (capability == ANDROID_REQUEST_AVAILABLE_CAPABILITIES_SYSTEM_CAMERA) {
+                return SystemCameraKind::SYSTEM_ONLY_CAMERA;
+            }
         }
     }
     return SystemCameraKind::PUBLIC;
-- 
2.25.1

